sr_gui_change_controllers/src/sr_gui_change_controllers/change_controllers.py

```diff
@@ -51,64 +58,179 @@ def __init__(self, context):
+            if 'la_' in self.avaliable_groups:
+                self._widget.lh_teach.hide()
+
+        # Disabiling until more than one control mode is avaliable for the arm
+        self._widget.ra_group.hide()
+        self._widget.la_group.hide()
+        # if 'ra_' not in self.avaliable_groups:
+        #     self._widget.ra_group.setDisabled(True)
+        # else:
+        #     self._widget.ra_traj.setIcon(self.CONTROLLER_OFF_ICON)
+        #     self._widget.ra_traj.toggled.connect(
+        #         self.teach_mode_button_toggled_ra)
+        #     self._ra_teach_buttons.append(self._widget.ra_traj)
+
+        #     self._widget.ra_pos.setIcon(self.CONTROLLER_OFF_ICON)
+        #     self._widget.ra_pos.toggled.connect(
+        #         self.teach_mode_button_toggled_ra)
+        #     self._ra_teach_buttons.append(self._widget.ra_pos)
+
+        # self._widget.ra_teach.setIcon(self.CONTROLLER_OFF_ICON)
+        # self._widget.ra_teach.toggled.connect(
+        #     self.teach_mode_button_toggled_ra)
+        # self._ra_teach_buttons.append(self._widget.ra_teach)
+
+        # if 'la_' not in self.avaliable_groups:
+        #     self._widget.la_group.setDisabled(True)
+        # else:
+        #     self._widget.la_traj.setIcon(self.CONTROLLER_OFF_ICON)
+        #     self._widget.la_traj.toggled.connect(
+        #         self.teach_mode_button_toggled_la)
+        #     self._la_teach_buttons.append(self._widget.la_traj)
+
+        #     self._widget.la_pos.setIcon(self.CONTROLLER_OFF_ICON)
+        #     self._widget.la_pos.toggled.connect(
+        #         self.teach_mode_button_toggled_la)
+        #     self._la_teach_buttons.append(self._widget.la_pos)
+
+        # self._widget.la_teach.setIcon(self.CONTROLLER_OFF_ICON)
+        # self._widget.la_teach.toggled.connect(
+        #     self.teach_mode_button_toggled_la)
+        # self._la_teach_buttons.append(self._widget.la_teach)
+
+        self._widget.information_box.clicked.connect(self.display_information)
+        self.confirm_current_control()
+
+    def find_avaliable_groups(self):
+        joint_states = rospy.wait_for_message('/joint_states', JointState)
+        robot_names = ["rh_", "lh_", "ra_", "la_"]
+        avaliable_robot_names = []
+        for robot_name in robot_names:
+            for joint in joint_states.name:
+                if joint.startswith(robot_name):
+                    if robot_name not in avaliable_robot_names:
+                        avaliable_robot_names.append(robot_name)
+        return avaliable_robot_names
+
+    def confirm_current_control(self):
+        """
+        @return: list of current controllers with associated data
+        """
+        success = True
+        list_controllers = rospy.ServiceProxy(
+            'controller_manager/list_controllers', ListControllers)
+        try:
+            resp1 = list_controllers()
+        except rospy.ServiceException:
+            success = False
```

Comment:
if this exception happens do you want to continue? with this code you will continue going and then check if controllers are running on an empty list? i think it would be better to fail here directly by raise the exception. then you can just continue checking the controller running without adding this extra logic